### PR TITLE
Bookstore: add reducers and actions

### PR DESCRIPTION
**Pull request by harlexkhal**
- Write your book's actions and reducer (in /src/redux/books/books.js) using the [ducks pattern](https://github.com/erikras/ducks-modular-redux).
- [x] Define action types for adding and removing a book.
- [x] Set the initial state to be an empty array of books.
- [x] Export Action Creators for your actions.
- [x] Write your reducer and export it as default
- [x] Define state changes for the actions that you created.
- [x] In case of unknown action - return the current state.

-  Write your categories actions and reducer (in /src/redux/categories/categories.js) using the [ducks pattern](https://github.com/erikras/ducks-modular-redux).
- [x] Define an action type for displaying "under construction" message.
- [x] Set the initial state to be an empty array of categories.
- [x] Export Action Creators for your actions.
- [x] Write your reducer and export it as default
- [x] For the check status action return a string "Under construction".
- [x] In case of unknown action - return the current state.

-  Configure the Redux Store (/src/redux/configureStore.js):
- [x] Importing the necessary methods from Redux.
- [x] Combine both reducers into a root reducer by using ` combineReducers`  function.
- [x] Use the root reducer to create your store.

### CODE REVIEW

**Review by dasileker — APPROVED** [review]
### **Hi @harlexkhal,** 

### **Congratulations! 🎉**

Your project is complete!  There is nothing else to say other than... it's time to merge it 🚀 : 
 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/redux/books/books.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
+export const addBook = ({ name, category }) => ({
```
Comment (nidalaa) [inline src/redux/books/books.js:1]: **[OPTIONAL] extra review comment **


Hello @harlexkhal  your project was selected as a random quality control check. ere is an additional comment that was missed by your code reviewer. 

You should use the dicks pattern for your reducers. Please take a look at the example here: https://github.com/erikras/ducks-modular-redux#example. You are missing 2 things, the first one:

- actions should be defined as `const` that later can be used in action creators and reducers.

File: src/redux/books/books.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  name,
+  category,
+});
+
+export const removeBook = (id) => ({
+  type: 'REMOVE_BOOK',
+  id,
+});
+
+const books = (state = [], action) => {
```
Comment (nidalaa) [inline src/redux/books/books.js:13]: **[OPTIONAL] extra review comment **


Hello @harlexkhal  your project was selected as a random quality control check. ere is an additional comment that was missed by your code reviewer. 

You should use the dicks pattern for your reducers. Please take a look at the example here: https://github.com/erikras/ducks-modular-redux#example. You are missing 2 things, the second one:

- this function should be exported as default and called `reducer`.

**Review by nidalaa — COMMENTED** [review]
Hello @harlexkhal your project was selected as a random quality control check.Check additional comments missed by your first reviewer :)

**Comment by harlexkhal** [comment]
@nidalaa Thanks for the review and thanks for the extra checks. all fixed on pull request #3. I'd like you to review my changes again before I merge. Thanks 💯

**Pull request merged**